app/services/ContractorDetailsService.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import com.google.inject.Inject
+import models.{AuthorisationRequest, ContractorDetailsViewModel, User}
+import models.mongo.{CisCYAModel, CisUserData, DatabaseError}
+import repositories.CisUserDataRepository
+import utils.Clock
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ContractorDetailsService @Inject()(cisUserDataRepository: CisUserDataRepository, clock: Clock){
+
+  def checkAccessContractorDetailsPage(taxYear: Int, user: User, employerRef: String)
+                                      (implicit executionContext: ExecutionContext): Future[Either[String, Option[CisUserData]]] = {
+    cisUserDataRepository.find(taxYear, employerRef, user).map(result => result match {
```

Comment:
match statement can be converted to pattern matching anonymous function (IntelliJ warning)